lib/gate/rails.rb

```diff
@@ -10,6 +10,11 @@ def self.included(base)
       base.send(:extend, ClassMethods)
     end
 
+    def self.configure(&block)
+      dynamic_class = "BaseSchema"
+      Object.const_set(dynamic_class, Class.new(Dry::Validation::Schema, &block))
```

Comment:
I wonder what's the difference, and why not this way:

`BaseSchema = Class.new(Dry::Validation::Schema, &block)`

or even:

`BaseSchema = Dry::Validation::Schema.call(&block)`
or
`Object.const_set("BaseSchema", Dry::Validation::Schema.call(&block))`

?

Reply:
From your propositions only the last one is valid(and I will use it!). Why? We are defining constant in method body. In Ruby, there is no possibility to create constant if there is a possibility an executing this part of code more than once.

Reply:
But a few minutes ago I noticed that gate without base configuration doesn't work proper. I'm on it.

Reply:
Done @jandudulski